numpy/core/tests/test_multiarray.py

```diff
@@ -7013,6 +7013,37 @@ def test_null_inside_ustring_array_is_truthy(self):
         assert_(a)
 
 
+class TestFormat(object):
+
+    def test_0d(self):
+        a = np.array(np.pi)
+        assert_equal('{:0.3g}'.format(a), '3.14')
```

Comment:
Is `assert_equal` the best way to compare strings? Maybe just `assert_(str1 == str2)` would be a bit cleaner.

Reply:
Yes, because then it shows the two values if they differ.

Reply:
OK.

Reply:
In pytest you can just `assert str1 == str2` and you will get values if there is a failure. 

```

def test_failure():
    a = 'apple'
    b = 'banana'
    assert a == b
```

Output is 

```
================================== FAILURES ===================================
________________________________ test_failure _________________________________

    def test_failure():
        a = 'apple'
        b = 'banana'
>       assert a == b
E       AssertionError: assert 'apple' == 'banana'
E         - apple
E         + banana

test_util.py:20: AssertionError
========================== 1 failed in 0.24 seconds ===========================
```

Reply:
For consistency, we should probably use the same form we use in most of the test suite, which gives:

```
>>> np.testing.assert_equal('a', 'b')
AssertionError: 
Items are not equal:
 ACTUAL: 'a'
 DESIRED: 'b'
```

AFAIK, the switch to `pytest` was recent, and we were previously using `nose`

Reply:
We are still using nose, the pytest conversion is still in progress and I don't think I will get back to it until 1.15.